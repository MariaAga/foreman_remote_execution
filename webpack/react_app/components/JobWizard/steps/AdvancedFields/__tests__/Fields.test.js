import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import {
  EffectiveUserField,
  TimeoutToKillField,
  PasswordField,
  KeyPassphraseField,
  EffectiveUserPasswordField,
  ConcurrencyLevelField,
  TimeSpanLevelField,
  TemplateInputsFields,
} from '../Fields';

const fixtures = {
  'renders ': { value: 'some value', setValue: jest.fn() },
};

describe('Fields', () => {
  describe('EffectiveUserField', () =>
    testComponentSnapshotsWithFixtures(EffectiveUserField, fixtures));

  describe('TimeoutToKillField', () =>
    testComponentSnapshotsWithFixtures(TimeoutToKillField, fixtures));

  describe('PasswordField', () =>
    testComponentSnapshotsWithFixtures(PasswordField, fixtures));

  describe('KeyPassphraseField', () =>
    testComponentSnapshotsWithFixtures(KeyPassphraseField, fixtures));

  describe('EffectiveUserPasswordField', () =>
    testComponentSnapshotsWithFixtures(EffectiveUserPasswordField, fixtures));

  describe('ConcurrencyLevelField', () =>
    testComponentSnapshotsWithFixtures(ConcurrencyLevelField, fixtures));

  describe('TimeSpanLevelField', () =>
    testComponentSnapshotsWithFixtures(TimeSpanLevelField, fixtures));

  describe('TemplateInputsFields', () =>
    testComponentSnapshotsWithFixtures(TemplateInputsFields, fixtures));
});
