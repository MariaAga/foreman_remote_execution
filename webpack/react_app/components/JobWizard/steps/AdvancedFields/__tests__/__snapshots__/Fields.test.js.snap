// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields ConcurrencyLevelField renders  1`] = `
<FormGroup
  fieldId="concurrency-level"
  helperTextInvalid="Concurrency level has to be a number, and at least 1"
  label="Concurrency level"
  labelIcon={
    <Popover
      aria-label="help-text"
      bodyContent="Run at most N tasks at a time. If this is set and proxy batch triggering is enabled, then tasks are triggered on the smart proxy in batches of size 1."
    >
      <button
        aria-label="More info for name field"
        className="pf-c-form__group-label-help"
        onClick={[Function]}
      >
        <HelpIcon
          color="currentColor"
          noVerticalAlign={true}
          size="sm"
        />
      </button>
    </Popover>
  }
  validated="default"
>
  <TextInput
    id="concurrency-level"
    onChange={[Function]}
    placeholder="For example: 1, 2, 3, 4, 5..."
    type="number"
    validated="default"
    value="some value"
  />
</FormGroup>
`;

exports[`Fields EffectiveUserField renders  1`] = `
<FormGroup
  fieldId="effective-user"
  label="Effective user"
  labelIcon={
    <Popover
      aria-label="help-text"
      bodyContent="A user to be used for executing the script. If it differs from the SSH user, su or sudo is used to switch the accounts."
    >
      <button
        aria-label="More info for name field"
        className="pf-c-form__group-label-help"
        onClick={[Function]}
      >
        <HelpIcon
          color="currentColor"
          noVerticalAlign={true}
          size="sm"
        />
      </button>
    </Popover>
  }
>
  <TextInput
    id="effective-user"
    onChange={[Function]}
    type="text"
    value="some value"
  />
</FormGroup>
`;

exports[`Fields EffectiveUserPasswordField renders  1`] = `
<FormGroup
  fieldId="effective-user-password"
  label="Effective user password"
  labelIcon={
    <Popover
      aria-label="help-text"
      bodyContent="Effective user password is only applicable for SSH provider. Other providers ignore this field. Password is stored encrypted in DB until the job finishes. For future or recurring executions, it is removed after the last execution."
    >
      <button
        aria-label="More info for name field"
        className="pf-c-form__group-label-help"
        onClick={[Function]}
      >
        <HelpIcon
          color="currentColor"
          noVerticalAlign={true}
          size="sm"
        />
      </button>
    </Popover>
  }
>
  <TextInput
    id="effective-user-password"
    onChange={[Function]}
    placeholder="*****"
    type="password"
    value="some value"
  />
</FormGroup>
`;

exports[`Fields KeyPassphraseField renders  1`] = `
<FormGroup
  fieldId="key-passphrase"
  label="Private key passphrase"
  labelIcon={
    <Popover
      aria-label="help-text"
      bodyContent="Key passphrase is only applicable for SSH provider. Other providers ignore this field. Passphrase is stored encrypted in DB until the job finishes. For future or recurring executions, it is removed after the last execution."
    >
      <button
        aria-label="More info for name field"
        className="pf-c-form__group-label-help"
        onClick={[Function]}
      >
        <HelpIcon
          color="currentColor"
          noVerticalAlign={true}
          size="sm"
        />
      </button>
    </Popover>
  }
>
  <TextInput
    id="key-passphrase"
    onChange={[Function]}
    placeholder="*****"
    type="password"
    value="some value"
  />
</FormGroup>
`;

exports[`Fields PasswordField renders  1`] = `
<FormGroup
  fieldId="password"
  label="Password"
  labelIcon={
    <Popover
      aria-label="help-text"
      bodyContent="Password is stored encrypted in DB until the job finishes. For future or recurring executions, it is removed after the last execution."
    >
      <button
        aria-label="More info for name field"
        className="pf-c-form__group-label-help"
        onClick={[Function]}
      >
        <HelpIcon
          color="currentColor"
          noVerticalAlign={true}
          size="sm"
        />
      </button>
    </Popover>
  }
>
  <TextInput
    id="password"
    onChange={[Function]}
    placeholder="*****"
    type="password"
    value="some value"
  />
</FormGroup>
`;

exports[`Fields TimeSpanLevelField renders  1`] = `
<FormGroup
  fieldId="time-span"
  helperTextInvalid="Concurrency level has to be a number, and at least 1"
  label="Time span"
  labelIcon={
    <Popover
      aria-label="help-text"
      bodyContent="Distribute execution over N seconds. If this is set and proxy batch triggering is enabled, then tasks are triggered on the smart proxy in batches of size 1."
    >
      <button
        aria-label="More info for name field"
        className="pf-c-form__group-label-help"
        onClick={[Function]}
      >
        <HelpIcon
          color="currentColor"
          noVerticalAlign={true}
          size="sm"
        />
      </button>
    </Popover>
  }
  validated="default"
>
  <TextInput
    id="time-span"
    min={1}
    onChange={[Function]}
    placeholder="For example: 1, 2, 3, 4, 5..."
    type="number"
    validated="default"
    value="some value"
  />
</FormGroup>
`;

exports[`Fields TimeoutToKillField renders  1`] = `
<FormGroup
  fieldId="timeout-to-kill"
  helperTextInvalid="Timeout to kill has to be a number"
  label="Timeout to kill"
  labelIcon={
    <Popover
      aria-label="help-text"
      bodyContent="Time in seconds from the start on the remote host after which the job should be killed."
    >
      <button
        aria-label="More info for name field"
        className="pf-c-form__group-label-help"
        onClick={[Function]}
      >
        <HelpIcon
          color="currentColor"
          noVerticalAlign={true}
          size="sm"
        />
      </button>
    </Popover>
  }
  validated="default"
>
  <TextInput
    id="timeout-to-kill"
    onChange={[Function]}
    placeholder="For example: 1, 2, 3, 4, 5..."
    type="number"
    validated="default"
    value="some value"
  />
</FormGroup>
`;
